packages/editor/src/components/post-publish-panel/maybe-tags-panel.js

```diff
@@ -34,38 +34,53 @@ const TagsPanel = () => {
 };
 
 const MaybeTagsPanel = () => {
-	const { hasTags, isPostTypeSupported } = useSelect( ( select ) => {
-		const postType = select( editorStore ).getCurrentPostType();
-		const tagsTaxonomy = select( coreStore ).getEntityRecord(
-			'root',
-			'taxonomy',
-			'post_tag'
-		);
-		const _isPostTypeSupported = tagsTaxonomy?.types?.includes( postType );
-		const areTagsFetched = tagsTaxonomy !== undefined;
-		const tags =
-			tagsTaxonomy &&
-			select( editorStore ).getEditedPostAttribute(
-				tagsTaxonomy.rest_base
+	const { postHasTags, siteHasTags, isPostTypeSupported } = useSelect(
+		( select ) => {
+			const postType = select( editorStore ).getCurrentPostType();
+			const tagsTaxonomy = select( coreStore ).getEntityRecord(
+				'root',
+				'taxonomy',
+				'post_tag'
 			);
-		return {
-			hasTags: !! tags?.length,
-			isPostTypeSupported: areTagsFetched && _isPostTypeSupported,
-		};
-	}, [] );
-	const [ hadTagsWhenOpeningThePanel ] = useState( hasTags );
+			const _isPostTypeSupported =
+				tagsTaxonomy?.types?.includes( postType );
+			const areTagsFetched = tagsTaxonomy !== undefined;
+			const tags =
+				tagsTaxonomy &&
+				select( editorStore ).getEditedPostAttribute(
+					tagsTaxonomy.rest_base
+				);
+			const siteTags = select( coreStore ).getEntityRecords(
```

Comment:
Hm, there is no point in check if the site has tags, if the current post type does not support them.

Reply:
That's a good point. You can do a conditional query `_isPostTypeSupported ? !! getEntityRecords()?.length : false`.

The same applies to the categories; there, you can probably combine the `useSelect` hooks into one.